fixed jalr when both oporands are the same register

idoheinemann opened the pull request:
when you would perform an action such as `jalr x4, x4`, it would not jump because it would set the value of x4 to the current location and the set pc to x4, which is now the current pc. this code fixes that problem

idoheinemann committed:
fixed jalr when both oporands are the same register

when you would perform an action such as `jalr x4, x4`, it would not jump because it would set the value of x4 to the current location and the set pc to x4, which is now the current pc. this code pervents it from happening

erikik8090 commented:
Hi,
Thanks for your contribution!
Could you add a test for your bug? A simple test case to make sure the bug doesn't return. Should fail at the current version, and succeed with your pull request.

idoheinemann committed:
added a test for `jalr reg1, reg1` scenarios

added a test for jalr instruction calling where the source register and destination register are the same
such as `jalr x3, x3`

idoheinemann committed:
removed double brackets
